.github/workflows/rspec_rubocop.yml

```diff
@@ -0,0 +1,77 @@
+# This workflow will download a prebuilt Ruby version, install dependencies and run tests with Rake
+# For more information see: https://github.com/marketplace/actions/setup-ruby-jruby-and-truffleruby
+
+name: Ruby
+
+on: [pull_request, push]
+
+permissions:
+  contents: read
+
+env:
+  CC: ccache gcc
+  CXX: ccache g++
+  THRIFT_SHA256SUM: 2289d02de6e8db04cbbabb921aeb62bfe3098c4c83f36eec6c31194301efa10b  thrift-0.10.0.tar.gz
+  THRIFT_URL: http://archive.apache.org/dist/thrift/0.10.0/thrift-0.10.0.tar.gz
+  THRIFT_FILE: thrift-0.10.0.tar.gz
+  THRIFT_DIR: thrift-0.10.0
+  DEBIAN_FRONTEND: noninteractive # for apt to be noninteractive
+  ImageOS: ubuntu20
+
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    container: ubuntu:focal
+    strategy:
+      matrix:
+        include: # use bundler 2.3 for ruby versions < 2.6 (https://bundler.io/compatibility.html)
+        - ruby-version: 2.5
+          bundler-version: 2.3
+        - ruby-version: 2.6
+          bundler-version: latest
+        - ruby-version: 2.7
+          bundler-version: latest
+        # - ruby-version: 3.0
+        #   bundler-version: latest
+        # - ruby-version: 3.1
+        #   bundler-version: latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install dependencies
+      run: |
+        apt-get update
+        apt-get install -y \
+          automake \
+          bison \
+          flex \
+          g++ \
+          git \
+          libboost-all-dev \
+          libevent-dev \
+          libssl-dev \
+          libtool \
+          make \
+          pkg-config \
+          cmake \
+          ccache \
+          wget \
+          ruby-dev
```

Comment:
This package is needed because bundler is failing to build the gem with the error:

```
/opt/hostedtoolcache/Ruby/2.5.9/x64/lib/ruby/2.5.0/mkmf.rb:456:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
```

You can see more info about the error [here](https://github.com/santiagorodriguez96/sparsam/actions/runs/4244187738/jobs/7377941504).

Reply:
Hmm interesting point actually 🤔 

It seems that `ruby-dev` package consists on `Header files for compiling extension modules for Ruby` so I wonder if using `ruby2.7-dev` – which is the one that installs with `Ubuntu Focal` – would affect the use of other Rubies.

Maybe there's a another workaround for this error...

Reply:
Okay so I was able to get it working without having to install `ruby-dev` – see 84130a14808f5d5a56013da36b839633d4b02169.

After trying a bunch of stuff, I realized that [`setup-ruby` requires `libyaml-dev` and `libgmp-dev` in the runner image in order to work](https://github.com/ruby/setup-ruby#using-self-hosted-runners), which seems to de the trick – you can see the build working [here](https://github.com/santiagorodriguez96/sparsam/commit/84130a14808f5d5a56013da36b839633d4b02169).